src/apps/api/views/mapping.py

```diff
@@ -1,83 +1,128 @@
+                    field = tol_field if standard == 'tol' else str()
+
+                # Ignore if 'field' is an empty string
+                if field:
+                    output_list.append(field)
                             
-            output.append(output_dict)
-
-            result = finish_request(template=output, error=None, num_found=len(output[0]),return_http_response=False)
+            return output_list
         else:
-            result =  finish_request(template=None, error=None, num_found=None, return_http_response=False)
+            return list()
 
-    # Removes slashes from output
-    result = json.loads(result)
-    return result
+def get_mapped_data_from_sample_data(standard, sample_data):
+    # COPO defined fields are fields that are lowercase or camel case
+    copo_defined_fields =  Sample().get_all_sample_fields(has_copo_defined_fields=True, has_project_defined_fields=False)
 
-def get_mapping(request):
-    return_type = request.GET.get('return_type', "json").lower()
+    if standard not in STANDARDS:
+        return None
     
-    default_value = d_utils.convertListToString(STANDARDS)
-    standard = request.GET.get('standard', default_value)
-    standard_list = d_utils.convertStringToList(standard)
-
-    if return_type == "csv":
-        return HttpResponse(content="Not Implemented")
-    
-    standard_data = get_standard_data(standard_list=standard_list)
-
-    return HttpResponse(jsonb.dumps(standard_data), content_type="application/json")
-
-
-def get_mapping_for_manifest_type(request, manifest_type):
-    return_type = request.GET.get('return_type', "json").lower()
-
-    default_value = d_utils.convertListToString(STANDARDS)
-    standard = request.GET.get('standard', default_value)
-    standard_list = d_utils.convertStringToList(standard) # Split the 'standard' string into a list
+    with open(STANDARDS_MAPPING_FILE_PATH) as f:
+        output_list = list()
+        standard_mapped_data = json.load(f)
+        standard_mapped_data = standard_mapped_data[0]
+        standard_fields = get_standard_fields(standard, standard_mapped_data)
+       
+        for sample in sample_data:
+            output_dict = dict()
+
+            for tol_field, value in sample.items():
+                if standard != 'tol' and tol_field in list(standard_mapped_data.keys()):
+                    field = standard_mapped_data[tol_field].get(standard, str()).get('field', str())
+                else:
+                    field = tol_field if standard == 'tol' else str()
+               
+                if field in standard_fields:
+                    # Proceed if the 'field' is a standard field
+                    pass
+                else:
+                    # If field is defined by COPO, add it to 'output_dict'
+                    if tol_field in copo_defined_fields or tol_field.lower().startswith('copo'):
+                        field = tol_field if tol_field.lower().startswith('copo') else f'copo_{tol_field}'
+
```

Comment:
```
if tol_field.lower().startswith('copo'):
      field = tot_field
elif tol_field in cope_defined_fields:
      field = f'copo_{tol_field}' 
```

it can avoid over checking "tol_field.lower().startswith('copo')".  Less code, more readable and lazy programming